tests/model_fields/tests.py

```diff
@@ -661,6 +689,15 @@ def test_backend_range_validation(self):
             }
             with self.assertRaisesMessage(ValidationError, expected_message):
                 instance.full_clean()
+            try:
+                with transaction.atomic():
+                    instance.save()
+            except (DataError, IntegrityError):
+                pass  # backends such as Postgres don't fail silently
+            else:
+                qs = self.model.objects.filter(value__lte=min_value)
```

Comment:
What about the idea of `supports_strict_insert_update` as a feature flag. The base value would be `True`, SQLite would be `False` and MySQL would have a `cached_property`. This would allow the  tests to pass while being deterministic of `STRICT_TRANS_TABLES` and `STRICT_ALL_TABLES` settings. As stated in [#15940](https://code.djangoproject.com/ticket/15940) strict mode is enabled by default for new installs of MySQL 5.6+.

This should allow for back porting to the 1.8.x branch and could also be extended to a deprecation warning into 1.9.x for MySQL installs that don't have strict mode enabled. When Django makes strict mode the default for MySQL we can then replace the `cached_property` with a constant and close [#15940](https://code.djangoproject.com/ticket/15940).